cloud/previews/previews.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package previews
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/terramate-io/terramate/cloud"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/errors"
+)
+
+const (
+	// PreviewStatusAffected is the status for a stack that is affected in a PR
+	PreviewStatusAffected = "affected"
+	// PreviewStatusPending is the status for a stack that is selected in a preview run
+	PreviewStatusPending = "pending"
+)
+
+// RunContext is the context for a run
+type RunContext struct {
+	Stack *config.Stack
+	Cmd   []string
+}
+
+// CreatePreviewOpts is the options for the CreatePreview function
+type CreatePreviewOpts struct {
+	Runs            []RunContext
+	AffectedStacks  map[string]*config.Stack
+	OrgUUID         cloud.UUID
+	UpdatedAt       int64
+	Technology      string
+	TechnologyLayer string
+	Repository      string
+	DefaultBranch   string
+	ReviewRequest   *cloud.DeploymentReviewRequest
+	Metadata        *cloud.DeploymentMetadata
+}
```

Comment:
this is a mix of required and optional fields so I would not call it `...Opts`.
I think it's better to pass the required information explicitly as function parameters and then this `Opt` type can be used for options.

Reply:
main motivation to add that struct was to have a smaller arg list. I would lean towards removing the struct all together and passing all of that information as function args instead of differentiating between optional/mandatory args because it's not meaningful IMO to mark optional/non-optional args. What do you think?